tests/app/questionnaire/test_navigation.py

```diff
@@ -662,3 +663,363 @@ def test_build_navigation_returns_navigation_when_schema_navigation_is_true(self
 
         # Then
         self.assertIsNotNone(nav_menu)
+
+    def test_build_navigation_summary_link_hidden_when_no_sections_completed(self):
+        survey = load_schema_file("test_navigation.json")
+        metadata = {
+            "eq_id": '1',
+            "collection_exercise_sid": '999',
+            "form_type": "some_form"
+        }
+
+        navigation = Navigation(survey, AnswerStore(), metadata)
+
+        confirmation_link = {
+            'link_name': 'Summary',
+            'highlight': False,
+            'repeating': False,
+            'completed': False,
+            'link_url': Location('summary-group', 0, 'summary').url(metadata)
+        }
+
+        self.assertNotIn(confirmation_link, navigation.build_navigation('property-details', 0))
+
+    def test_build_navigation_summary_link_hidden_when_not_all_sections_completed(self):
+        survey = load_schema_file("test_navigation.json")
+        metadata = {
+            "eq_id": '1',
+            "collection_exercise_sid": '999',
+            "form_type": "some_form"
+        }
+
+        completed_blocks = [
+            Location('property-details', 0, 'insurance-type'),
+            Location('property-details', 0, 'insurance-address'),
+            Location('property-details', 0, 'property-interstitial'),
+            Location('house-details', 0, 'house-type'),
+            Location('multiple-questions-group', 0, 'household-composition'),
+        ]
+
+        navigation = Navigation(survey, AnswerStore(), metadata, completed_blocks)
+
+        confirmation_link = {
+            'link_name': 'Summary',
+            'highlight': False,
+            'repeating': False,
+            'completed': False,
+            'link_url': Location('summary-group', 0, 'summary').url(metadata)
+        }
+
+        self.assertNotIn(confirmation_link, navigation.build_navigation('property-details', 0))
```

Comment:
Is it also worth checking the number of links in the navigation for all of these `not in` tests